spanner_orm/admin/column.py

```diff
@@ -39,8 +41,36 @@ def nullable(self) -> bool:
 
   def field_type(self) -> Type[field.FieldType]:
     for field_type in field.ALL_TYPES:
-      if self.spanner_type == field_type.ddl():
+      len = _get_type_len(self.spanner_type)
+      contain_commit_timestamp_option = _get_commit_timestamp_option(
+          self.spanner_type)
+      actual_type = field_type.ddl()
+      if len:
+        actual_type = actual_type.replace('(MAX)', f'({len})')
+      if contain_commit_timestamp_option:
+        actual_type = f'{actual_type}{allow_commit_timestamp_option}'
+      if self.spanner_type == actual_type:
         return field_type
 
     raise error.SpannerError('No corresponding Type for {}'.format(
         self.spanner_type))
+
+
+def _get_type_len(spanner_type: str) -> int:
```

Comment:
this method should be in field_type instead.

Reply:
Done. Removed this method